### ISSUE TITLE
Add describe_transform_job() to the SageMaker Session class

### ISSUE THREAD

**Issue by alex23lemm** [aws/sagemaker-python-sdk#1502]
**Is your feature request related to a problem? Please describe.**
A SageMaker Session object comes with the following two methods which allow me to retrieve the status of training and AutoML jobs: `describe_training_job()`, `describe_auto_ml_job()`. Example:

``` python
import sagemaker

session = sagemaker.Session()
session.describe_training_job("[YOUR_JOB_NAME]")["TrainingJobStatus"]
```
However, if I like to get the status of a batch transform job (like, e.g., a batch interence job) i**n the same workflow**, I need to create an extra low-level SageMaker client using boto3:

``` python
import boto3

sm = boto3.client("sagemaker")
sm.describe_transform_job(TransformJobName = "[YOUR_JOB_NAME]")["TransformJobStatus"]
```

**Describe the solution you'd like**
Even though the above workflow works it could become much more concise and smoother by adding a `describe_transform_job()` method to the SageMaker Session class.  This would eliminate the need to use the extra low-level SageMaker client in the same workflow.

**Comment by laurenyu** [comment]
thanks for the feature request! this was just addressed in #1507, and should be released tomorrow

**Comment by alex23lemm** [comment]
Thanks @laurenyu for the feedback and for adding the request so quickly to the next release!

**Comment by laurenyu** [comment]
this has now been released in [v1.58.4](https://github.com/aws/sagemaker-python-sdk/releases/tag/v1.58.4). thanks to @ChuyangDeng for jumping on the implementation so quickly :)
